website/docs/topics/long_contexts.md

```diff
@@ -0,0 +1,365 @@
+# Handling Long Context Conversations with Transform Messages
+
+Why do we need to handle long contexts? The problem arises from several constraints and requirements:
+
+1. Token limits: LLMs have token limits that restrict the amount of textual data they can process. If we exceed these limits, we may encounter errors or incur additional costs. By preprocessing the chat history, we can ensure that we stay within the acceptable token range.
+
+2. Context relevance: As conversations progress, retaining the entire chat history may become less relevant or even counterproductive. Keeping only the most recent and pertinent messages can help the LLMs focus on the most crucial context, leading to more accurate and relevant responses.
+
+3. Efficiency: Processing long contexts can consume more computational resources, leading to slower response times.
+
+## Transform Messages Capability
+
+The `TransformMessages` capability is designed to modify incoming messages before they are processed by the LLM agent. This can include limiting the number of messages, truncating messages to meet token limits, and more.
+
+### Installation and Setup
```

Comment:
I would suggest a different structure of the coding part. 
* **Part 1**: explore and understand transformations (include the examples you already have).
* **Part 2**: transformations with agents, this can have the following subparts:
   a) Setting up the stage: init ProxyAgent, config list
   b) Equip agent with Transform Messages capability: write test function and explain, run for agent without the capability, run for agent with the capability.
   c) Custom Transformation: write the new transformation and explain it, run the example